From 69c5804f19bff9df340c7335082a055663d374e9 Mon Sep 17 00:00:00 2001
From: Lulu Cheng <chenglulu@loongson.cn>
Date: Thu, 13 Apr 2023 19:24:38 +0800
Subject: [PATCH 038/303] LoongArch: Remove the definition of the macro
 LOGICAL_OP_NON_SHORT_CIRCUIT under the architecture and use the default
 definition instead.

In some cases, setting this macro as the default can reduce the number of conditional
branch instructions.

gcc/ChangeLog:

	* config/loongarch/loongarch.h (LOGICAL_OP_NON_SHORT_CIRCUIT): Remove the macro
	definition.

Signed-off-by: Peng Fan <fanpeng@loongson.cn>
Signed-off-by: ticat_fp <fanpeng@loongson.cn>
---
 gcc/config/loongarch/loongarch.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/gcc/config/loongarch/loongarch.h b/src/gcc/config/loongarch/loongarch.h
index 4b3de812c..f198e3f5b 100644
--- a/src/gcc/config/loongarch/loongarch.h
+++ b/src/gcc/config/loongarch/loongarch.h
@@ -836,7 +836,6 @@ typedef struct {
    1 is the default; other values are interpreted relative to that.  */
 
 #define BRANCH_COST(speed_p, predictable_p) loongarch_branch_cost
-#define LOGICAL_OP_NON_SHORT_CIRCUIT 0
 
 /* Return the asm template for a conditional branch instruction.
    OPCODE is the opcode's mnemonic and OPERANDS is the asm template for
-- 
2.33.0

